cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -83,6 +91,92 @@ public void download(String url, Path target) {
     }
   }
 
+  /**
+   * Downloads a file while showing a progress bar
+   *
+   * @param url the url to download
+   * @param target Path of the target directory
+   * @param response {@link HttpResponse} to use
+   */
+  private void downloadFileWithProgressBar(String url, Path target, HttpResponse<InputStream> response) throws IOException {
+
+    String contentLength = response.headers().firstValue("content-length").orElse("undefined");
+    if (contentLength.equals("undefined")) {
+      throw new IllegalStateException("Content-Length was not provided by download source: " + url);
+    }
+
+    long size = Long.parseLong(contentLength);
+
+    byte[] data = new byte[1024];
+    boolean fileComplete = false;
+    int count;
+    ProgressBarBuilder pbb = getProgressBarBuilder(size, "Downloading");
+
+    try (InputStream body = response.body();
+        FileOutputStream fileOutput = new FileOutputStream(target.toFile());
+        BufferedOutputStream bufferedOut = new BufferedOutputStream(fileOutput, 1024);
+        ProgressBar pb = pbb.build()) {
+      while (!fileComplete) {
+        count = body.read(data, 0, 1024);
+        if (count <= 0) {
+          fileComplete = true;
+        } else {
+          bufferedOut.write(data, 0, count);
+          pb.stepBy(count);
+        }
+      }
```

Comment:
Nice solution. What we IMHO will need is some abstraction as in tests or in the GUI the `ProgressBar` implementation has to be exchanged with something else.
This abstraction should IMHO be provided in (Abstract)IdeContext.

Reply:
I've refactored the ProgressBar to IdeContext now.

Reply:
IMHO the updates we have done together will resolve all this. I will wait until those are pushed and then we will most likely be able to merge this PR.